### ISSUE TITLE
Role inheritance

### ISSUE THREAD

**Issue by JortvD** [JortvD/meta#3]
It might be that I'm thinking to far into the future. The idea is that roles (sets of permissions) can be created for nodes in the machine tree (see concepts). If you create a role for a parent node you can then also assign that role to users for it children nodes. Of course the parent node might have more (or less) functionality than its children and thus for the children it needs to be determined what permissions apply. Does anyone have an opinion on this and maybe know some fallacies that would make this system very hard to implement?

**Comment by theotterlord** [comment]
So, would each node have a role or just the users? If just the users, we'll have to make sure that user credentials are passed on, if one node calls another for some reason.

**Comment by JortvD** [comment]
You can apply roles to users (as it is a set of permissions basically). My idea was that this can be done globally or for hosting sites for example you should also be able to apply roles that are more local, e.g. just a node or even the child node of a node. This way hosting sites can limit certain nodes to certain users.
Passing on of credentials might be needed for users that directly interact with the machine instance, but my thought was that for normal usage the user interact with the main interface, which can then verify the authenticity of the user and pass on the message to the node.

**Comment by theotterlord** [comment]
Thanks for the explaination. 

This seems like a really good idea, and something that should probably be implemented early on. I can't think of anything that would make it difficult to implement, though I suppose it depends on what language we use, and how we implement it.

**Comment by JortvD** [comment]
I agree. As programming language I was thinking of using JS, with NodeJS handling the backend processes. I prefer JS because it is what I'm best at, it is one of the most well-known languages and it has a good ecosystem to begin from.

**Comment by theotterlord** [comment]
Yeah, I do a lot of JavaScript programming. It would be great if others wish to join as well, since as you said, it is one of the most well known programming languages.
